x/leverage/keeper/borrows.go

```diff
@@ -21,18 +22,34 @@ func (k Keeper) assertBorrowerHealth(ctx sdk.Context, borrowerAddr sdk.AccAddres
 	borrowed := k.GetBorrowerBorrows(ctx, borrowerAddr)
 	collateral := k.GetBorrowerCollateral(ctx, borrowerAddr)
 
-	value, err := k.TotalTokenValue(ctx, borrowed, types.PriceModeHigh)
+	// check health using collateral weight
+	borrowValue, err := k.TotalTokenValue(ctx, borrowed, types.PriceModeHigh)
 	if err != nil {
 		return err
 	}
-	limit, err := k.VisibleBorrowLimit(ctx, collateral)
+	borrowLimit, err := k.VisibleBorrowLimit(ctx, collateral)
 	if err != nil {
 		return err
 	}
-	if value.GT(limit.Mul(maxUsage)) {
+	if borrowValue.GT(borrowLimit.Mul(maxUsage)) {
 		return types.ErrUndercollaterized.Wrapf(
-			"borrowed: %s, limit: %s, max usage %s", value, limit, maxUsage)
+			"borrowed: %s, limit: %s, max usage %s", borrowValue, borrowLimit, maxUsage)
 	}
+
+	// check health using borrow factor
+	weightedBorrowValue, err := k.WeightedBorrowValue(ctx, borrowed, types.PriceModeHigh)
```

Comment:
No - borrowed value cannot ignore missing prices so both oracle and non-oracle errors must be returned.

The cases where oracle errors can be ignored are for collateral value, since as long as a borrower's _known_ collateral covers their borrow position, then adding any additional _unknown_ collateral could only increase their limit, but not decrease it and cause an unsafe position